Updated devevn_rhel8 doc with (#1252)
openshift/microshift: docs/devenv_rhel8.md

echo -e 'microshift\tALL=(ALL)\tNOPASSWD: ALL' | sudo tee /etc/sudoers.d/microshift
sudo dnf clean all -y
sudo dnf update -y

> sudo dnf install -y git cockpit make gcc selinux-policy-devel rpm-build bash-completion go-toolset

Reviewer: Why is this package needed? Also, if we change anything here, automation script need to be updated.

Author: It installs golang, otherwise `make` won't build the microshift executable, it throws error about `go` not present on the system (atleast in RHEL 8.7)

Reviewer: Right. We missed reverting this line when we moved to go installation from RPM. The command used in the automation is the correct one (see https://github.com/openshift/microshift/blob/main/scripts/devenv-builder/configure-vm.sh#L47) 

```suggestion
sudo dnf install -y git cockpit make golang selinux-policy-devel rpm-build bash-completion
```

Author: Done.

### RPM Packages
Run make command with the `rpm` or `srpm` argument in the top-level directory. 
```bash

> make clean

Reviewer: We need clean because these instructions are reused in the automation script and it should be idempotent. If clean is not present, the RPM install command may fail because it'll pick up RPMs from more than one build.

Author: I was following the instructions in the doc manually (as i believe most of the folks trying to setup their local environment). In the `Executable` section above, make builds the microshift executable, but `make clean` in this section deletes that, and you need to make it again. I think having a specific make target for cleaning executable and RPMS would be better i think. what do you think?

Reviewer: I would not add a separate target just for deleting microshift executable. Note that RPM build rebuilds its own executable in a different location.
What about moving the 'make clean' before 'make' in the previous section?

Author: yeah, that's reasonable. Moved make clean to Executable section.

```

To view all the available profiles, run `oc get --raw /debug/pprof`.

## Troubleshooting

> ### No valid repository ID for fast-datapath

Reviewer: ```suggestion
### No Valid Repository ID for OpenShift RPM Channels
```

Author: done

If you encounter following error message while enabling fast-datapath repo:

```

> # sudo subscription-manager repos \

Reviewer: Let's remove the command and leave the message alone. In general, the same problem may occur for rhocp channel enablement

Author: done.

>     --enable fast-datapath-for-rhel-8-$(uname -i)-rpms
Error: 'fast-datapath-for-rhel-8-x86_64-rpms' does not match a valid repository ID. Use "subscription-manager repos --list" to see valid repositories.
```

> Please make sure that your system is registered and attached to the `Red Hat Openshift Container Platform` subscription. Once the repo is enabled, you must see atleast following repo enabled.

Reviewer: ```suggestion
Make sure that your system is registered and attached to the `Red Hat Openshift Container Platform` or equivalent subscription. Once the repo is enabled, you must see at least the following repos enabled.
```

Author: done
